platform/macos/detect.py

```diff
@@ -269,8 +269,17 @@ def configure(env: "Environment"):
 
     env.Append(LINKFLAGS=["-rpath", "@executable_path/../Frameworks", "-rpath", "@executable_path"])
 
+    if env["metal"] and env["arch"] != "arm64":
+        print("Metal is only supported on arm64.")
```

Comment:
Any reason for not supporting it. x86 with AMD GPUs should have the same features as Apple Silicon, maybe apart from a few M3 specific features, which should not be used (at least without runtime checks and fallback) in any case. Or it just was not tested.

Reply:
There is a lot more complexity in MoltenVK to support non-Apple GPUs, including many work arounds, that can only be tested on actual hardware, that I don't have. I think it is reasonable to lean on MoltenVK to handle those complexities?

Reply:
I have not used any M3 specific features 👍🏻

Reply:
I think the macos builders that accessible on macstadium are cheapest on older hardware, so can you estimate how difficult to support non-Apple GPUs.

I have an old mac on hand so I was curious. Like.. the intel gpu or a external gpu on older macs.

Reply:
Whilst developing support for Metal, I've assumed Apple GPUs across all supported platforms (macOS, iOS and future visionOS and tvOS) and always intended MoltenVK to be the fallback for Intel hardware and non-Apple GPUs. There are numerous exceptions in MoltenVK to support Nvidia, AMD and integrated Intel GPUs, along with workarounds when behaviours differ, and we can defer to their vast experience, to support those hardware combinations.

This would be a significant challenge to develop and test without access to physical hardware (and a lot more time).

Reply:
I think it is fair to only support Apple silicon in the Metal backend for now.

Reply:
We need to make sure the fallback to MoltenVK happens for non-Apple Silicon hardware though. But that's not blocking for this PR and can be a follow-up.